src/theory/fp/fp_converter.cpp

```diff
@@ -12,23 +12,1155 @@
  ** \brief Conversion of floating-point operations to bit-vectors using symfpu.
  **/
 
+// theory.h Only needed for the leaf test
 #include "theory/fp/fp_converter.h"
+#include "theory/theory.h"
 
 #include <stack>
 
+#include "symfpu/core/add.h"
+#include "symfpu/core/classify.h"
+#include "symfpu/core/compare.h"
+#include "symfpu/core/convert.h"
+#include "symfpu/core/divide.h"
+#include "symfpu/core/fma.h"
+#include "symfpu/core/ite.h"
+#include "symfpu/core/multiply.h"
+#include "symfpu/core/packing.h"
+#include "symfpu/core/remainder.h"
+#include "symfpu/core/sign.h"
+#include "symfpu/core/sqrt.h"
+#include "symfpu/utils/numberOfRoundingModes.h"
+#include "symfpu/utils/properties.h"
+
+namespace symfpu {
+using namespace ::CVC4::theory::fp::symfpuSymbolic;
+
+#define CVC4_SYM_ITE_DFN(T)                                                \
```

Comment:
... and then there is this problem.  It is a macro because we need four copies of exactly the same code.  It would be great to do it with a template but that causes problems with template specialisation.  If you want I can explain at greater length but, at the moment, this is the best of a bad set of options.